test/lib/ufe/testContextOps.py

```diff
@@ -1476,5 +1466,31 @@ def testAssignExistingMaterialToSingleObject(self):
         cmds.undo()
         self.assertFalse(capsulePrim.HasAPI(UsdShade.MaterialBindingAPI))
 
+    @unittest.skipIf(os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') < '4010', 'Test only available in UFE preview version 0.4.10 and greater')
+    @unittest.skipUnless(Usd.GetVersion() >= (0, 21, 8), 'Requires CanApplySchema from USD')
+    def testGeomCoponentAssignment(self):
+        '''Duplicate a Maya cube to USD and then assign a material on a face.'''
+
+        cubeXForm, _ = cmds.polyCube(name='MyCube')
+        psPathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+
+        mayaUsd.lib.PrimUpdaterManager.duplicate(cmds.ls(cubeXForm, long=True)[0], psPathStr)
+
+        topPath = ufe.PathString.path(psPathStr + ',/' + cubeXForm + "/" + "top")
+        topItem = ufe.Hierarchy.createItem(topPath)
+        topSubset = UsdGeom.Subset(usdUtils.getPrimFromSceneItem(topItem))
+
+        self.assertEqual(topSubset.GetFamilyNameAttr().Get(), "componentTag")
+        self.assertFalse(topSubset.GetPrim().HasAPI(UsdShade.MaterialBindingAPI))
+
+        contextOps = ufe.ContextOps.contextOps(topItem)
+        cmd = contextOps.doOpCmd(['Assign New Material', 'USD', 'UsdPreviewSurface'])
+        self.assertIsNotNone(cmd)
+        ufeCmd.execute(cmd)
+
+        self.assertEqual(topSubset.GetFamilyNameAttr().Get(), "materialBind")
+        self.assertTrue(topSubset.GetPrim().HasAPI(UsdShade.MaterialBindingAPI))
```

Comment:
I guess it was discussed internally, but i'm not sure what is better: change the name of the set or create a new set? The family name, according to USD, is used to group tags that are related, so changing it changes the group-of-subset composed of with members with the same family name.

In this case we're the one generating the componentTag, but it feels bad to now have 5 side with componentTag family name and one with the material binding one.